python/pyspark/sql/session.py

```diff
@@ -600,6 +600,7 @@ def create(self) -> "SparkSession":
 
     _instantiatedSession: ClassVar[Optional["SparkSession"]] = None
     _activeSession: ClassVar[Optional["SparkSession"]] = None
+    _thread_local = local()
```

Comment:
Let's make this as an instance within session because one thread might have multiple sessions running